pages/[endpoint].tsx

```diff
@@ -0,0 +1,194 @@
+          loading={!endpoint || transactionsLoading}
+          transactions={
+            transactionsLoading
+              ? null
+              : (transactionsData?.chainQuery.transactionQuery?.transactions as
+                  | Transaction[]) ?? null
+          }
+        />
+      );
+    }
+  }, [
+    blocks,
+    blocksData?.chainQuery.blockQuery?.blocks,
+    blocksError,
+    blocksLoading,
+    endpoint,
+    transactionsData?.chainQuery.transactionQuery?.transactions,
+    transactionsError,
+    transactionsLoading,
+  ]);
+
+  return (
+    <main>
+      <Checkbox
+        label="Include blocks having any tx"
+        checked={excludeEmptyTxs}
+        onChange={() => setExcludeEmptyTxs(!excludeEmptyTxs)}
+      />
+      <SummaryCards blocks={blocks} />
+      <OffsetSwitch
+        olderHandler={olderHandler}
+        newerHandler={newerHandler}
+        disable={{
+          older: !endpoint || blocksLoading || transactionsLoading,
+          newer:
+            !endpoint || blocksLoading || transactionsLoading || offset < 1,
+        }}
+      />
+      <Pivot>
+        <PivotItem headerText="Blocks">{blockList}</PivotItem>
+        <PivotItem headerText="Transactions">{transactionList}</PivotItem>
+      </Pivot>
+    </main>
+  );
+}
+
+function SummaryCards({ blocks }: { blocks: Block[] | null }) {
+  if (blocks === null)
+    return <Cards interval={0} difficultyAverage={0} totalTxNumber={0} />;
+
+  const timestamps: Date[] = blocks.map(block => new Date(block.timestamp));
+
+  let interval = 0;
+  for (let i = 0; i < timestamps.length - 1; i++) {
+    interval += +timestamps[i] - +timestamps[i + 1];
+  }
+  interval /= (timestamps.length - 1) * 1000;
+
+  const difficulties = blocks.map(block => block.difficulty);
+  const difficultyAverage =
+    difficulties.reduce((d, sum) => d + sum, 0) / difficulties.length;
+
+  const txNumbers = blocks.map(block => block.transactions.length);
+  const totalTxNumber = txNumbers.reduce((a, b) => a + b, 0);
+  return (
+    <Cards
+      interval={interval}
+      difficultyAverage={difficultyAverage}
+      totalTxNumber={totalTxNumber}
+    />
+  );
+}
+
+function Cards({
+  interval,
+  difficultyAverage,
+  totalTxNumber,
+}: {
+  interval: number;
+  difficultyAverage: number;
+  totalTxNumber: number;
+}) {
+  return (
+    <div className="cards">
+      <div className="card" key="interval">
+        <strong>{interval.toFixed(ROUND_DIGITS)}</strong> sec
+        <p>Average interval in this page</p>
+      </div>
+      <div className="card" key="difficultyAverage">
+        <strong>{Math.floor(difficultyAverage).toLocaleString()}</strong>
+        <p>Average difficulty in this page</p>
+      </div>
+      <div className="card" key="total-tx-number">
+        <strong>{Math.floor(totalTxNumber).toLocaleString()}</strong>
+        <p>Total txs in this page</p>
+      </div>
+    </div>
+  );
+}
+
+export { getStaticProps, getStaticPaths };
```

Comment:
I think this project dosen't need to use SSR.
if the endpoint is what you need. you can get like this
`const {query:{endpoint}} = useRouter()`

Reply:
Actually, since one of the intents of this project is to be used as a static site embedded into Libplanet.Explorer, this is required. Also, the Next.js router actually needs to be waited for isReady (I presume that it's related to the server side rendering-first nature of Next.js) extracting the query has to be a bit cumbersome 😢)